src/main/java/org/opensearch/security/support/SecurityUtils.java

```diff
@@ -140,4 +149,52 @@ private static String resolveEnvVar(String envVarName, String mode, boolean bc,
             return bc ? Hasher.hash(envVarValue.toCharArray(), settings) : envVarValue;
         }
     }
+
+    // This is used to match host names - for e.g. *.example.local to dev.example.local
+    public static boolean matchesHostNamePatterns(WildcardMatcher hostMatcher, InetAddress address, String hostResolverMode) {
+        if (address == null || hostMatcher == null) {
+            return false;
+        }
+
+        // Only proceed with hostname checks if hostResolverMode is configured
+        if (hostResolverMode != null
+            && (hostResolverMode.equalsIgnoreCase("ip-hostname") || hostResolverMode.equalsIgnoreCase("ip-hostname-lookup"))) {
+            try {
+                List<String> valuesToCheck = new ArrayList<>(List.of(address.getHostAddress()));
+                final String hostName = address.getHostName();
+                valuesToCheck.add(hostName);
+                return valuesToCheck.stream().anyMatch(hostMatcher);
+            } catch (Exception e) {
+                log.warn("Failed to resolve hostname for {}: {}", address.getHostAddress(), e.getMessage());
+                return false;
+            }
+        }
+        return false;
+    }
+
+    // This is used to match IP addresses - for e.g. 192.168.0.1 to 192.168.0.0/16
+    public static boolean matchesIpAndCidrPatterns(ClientBlockRegistry<InetAddress> clientBlockRegistry, InetAddress address) {
+        if (address == null || clientBlockRegistry == null) {
+            return false;
+        }
+
+        AuthFailureListener authFailureListener = (AuthFailureListener) clientBlockRegistry;
+        final String hostAddress = address.getHostAddress();
+
+        return authFailureListener.getIgnoreHosts().stream().anyMatch(pattern -> {
+            // Handle CIDR patterns
+            if (pattern.indexOf('/') != -1) {
```

Comment:
I do not think that using `/` is sufficient for identifying CIDRs. The `ignore_hosts` property supports the whole wildcard matcher feature set which also supports specifying regexes using the `/regex/` syntax. This would also hit on these regexes.

Reply:
good catch. one of the options is to implement regex check for ipv4/ipv6 addresses to identify CIDRs specifically - will check other options and update the PR

Reply:
You can also keep it simple and get the last index of `/` as long as `/` is not the final char in the string